sqlserver/datadog_checks/sqlserver/schemas.py

```diff
@@ -0,0 +1,401 @@
+        super(Schemas, self).__init__(
+            check,
+            run_sync=True,
+            enabled=is_affirmative(config.schema_config.get('enabled', True)),
+            expected_db_exceptions=(),
+            # min collection interval is a desired collection interval for a check as a whole.
+            min_collection_interval=config.min_collection_interval,
+            dbms="sqlserver",
+            rate_limit=1 / float(collection_interval),
+            job_name="query-schemas",
+            shutdown_callback=self.shut_down,
+        )
+        base_event = {
+            "host": None,
+            "agent_version": datadog_agent.get_version(),
+            "dbms": "sqlserver",
+            "kind": "sqlserver_databases",
+            "collection_interval": collection_interval,
+            "dbms_version": None,
+            "tags": self._check.non_internal_tags,
+            "cloud_metadata": self._check._config.cloud_metadata,
+        }
+        self._data_submitter = SubmitData(self._check.database_monitoring_metadata, base_event, self._log)
+
+    def run_job(self):
+        self._collect_schemas_data()
+
+    def shut_down(self):
+        self._data_submitter.submit()
+
+    @tracked_method(agent_check_getter=agent_check_getter)
+    def _collect_schemas_data(self):
+        """Collects database information and schemas and submits to the agent's queue as dictionaries
+        schema dict
+        key/value:
+            "name": str
+            "id": str
+            "owner_name": str
+            "tables" : list of tables dicts
+                table
+                key/value:
+                    "id" : str
+                    "name" : str
+                    columns: list of columns dicts
+                        columns
+                        key/value:
+                            "name": str
+                            "data_type": str
+                            "default": str
+                            "nullable": bool
+                indexes : list of index dicts
+                    index
+                    key/value:
+                        "name": str
+                        "type": str
+                        "is_unique": bool
+                        "is_primary_key": bool
+                        "is_unique_constraint": bool
+                        "is_disabled": bool,
+                        "column_names": str
+                foreign_keys : list of foreign key dicts
+                    foreign_key
+                    key/value:
+                        "foreign_key_name": str
+                        "referencing_table": str
+                        "referencing_column": str
+                        "referenced_table": str
+                        "referenced_column": str
+                partitions: partition dict
+                    partition
+                    key/value:
+                        "partition_count": int
+        """
+        start_time = time.thread_time()
```

Comment:
i think we will want to use `time.time()` here

Reply:
mm yeah , I'll change.